src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py

```diff
@@ -890,9 +890,35 @@ def update_subnet(instance, resource_group_name, address_prefix=None, network_se
 
 # endregion
 
-def update_vpn_connection(instance, routing_weight=None):
+def update_vpn_connection(instance, resource_group_name, routing_weight=None, shared_key=None,
+                          tags=None, enable_bgp=None):
+
+    ncf = _network_client_factory()
+
     if routing_weight is not None:
         instance.routing_weight = routing_weight
+
+    if shared_key is not None:
+        instance.shared_key = shared_key
+
+    if tags is not None:
+        instance.tags = tags
+
+    if enable_bgp is not None:
+        instance.enable_bgp = enable_bgp
+
+    # TODO: Remove these when issue #1615 is fixed
+    instance.virtual_network_gateway1 = ncf.virtual_network_gateways.get(
+        resource_group_name, instance.virtual_network_gateway1.id.rsplit('/')[-1])
```

Comment:
In case #1615 won't be fixed soon... is it possible the gateway1 can be in a different resource group?